pkg/repositories/kuadrant/repository.go

```diff
@@ -53,3 +59,35 @@ func (r *kuadrantRepository) ListAPI(ctx context.Context) (*api.APIs, error) {
 
 	return &apis, nil
 }
+
+func (r *kuadrantRepository) GetAPIListState(ctx context.Context) (string, error) {
+	apiListStatus, err := r.getApiListStatus(ctx)
+	if err != nil {
+		return "", err
+	}
+	return apiListStatus.Data["fresh"], nil
+}
+
+func (r *kuadrantRepository) UpdateAPIListState(ctx context.Context, newHash string) error {
+	apiListStatus, err := r.getApiListStatus(ctx)
+	if err != nil {
+		return err
+	}
+
+	if apiListStatus.Data["hash"] == newHash && apiListStatus.Data["fresh"] != "false" {
+		apiListStatus.Data["fresh"] = "false"
+		if updateErr := r.client.Update(ctx, apiListStatus); updateErr != nil {
+			return updateErr
+		}
+	}
+	return nil
+}
+
+func (r *kuadrantRepository) getApiListStatus(ctx context.Context) (*v1.ConfigMap, error) {
+	apiListStatus := &v1.ConfigMap{}
+	namespacedName := k8stypes.NamespacedName{Namespace: kuadrantNamespace, Name: apiListStatusConfigMap}
+	if err := r.client.Get(ctx, namespacedName, apiListStatus); err != nil && errors.IsNotFound(err) {
+		return nil, err
+	}
+	return apiListStatus, nil
```

Comment:
Careful! Other errors will return a null api list status and null error.

Reply:
Copy/Pasta mistake. It doesn't even make sense that the status configMap was not found, it would always be a 5XX in terms of communication to the client.